docs/specs/openapi.yaml

```diff
@@ -0,0 +1,335 @@
+                            total: 1
+                          spawn-time: "2024-12-12T12:50:29.743512342+08:00"
+                          ready-time: "2024-12-12T12:50:30.751092759+08:00"
+                      ready: true
+                      spawn-time: "2024-12-12T12:50:29.743570175+08:00"
+                      ready-time: "2024-12-12T12:50:30.751094634+08:00"
+        '400':
+          description: Bad Request - invalid timeout parameter.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        '404':
+          description: Not Found - change ID not found.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        '504':
+          description: Gateway Timeout - timeout waiting for change.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /v1/services:
+    get:
+      summary: Get services
+      description: Fetch information about specific services (or all of them), ordered by service name.
+      tags:
+        - services
+      parameters:
+        - in: query
+          name: names
+          description: The names of the services to get, a comma-separated string. If empty, get all services.
+          schema:
+            type: string
+          examples:
+            one:
+              value: svc1
+              summary: one service name
+            two:
+              value: svc1,svc2
+              summary: multiple services names, comma separated
+      responses:
+        '200':
+          description: List of services
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetServices"
+              examples:
+                list all services:
+                  value:
+                    - type: sync
+                      status-code: 200
+                      status: OK
+                      result:
+                        - name: "svc1"
+                          startup: "enabled"
+                          current: "active"
+                          current-since: "2024-12-11T20:10:33.381993977+08:00"
+                        - name: "svc2"
+                          startup: "disabled"
+                          current: "stopped"
+    post:
+      summary: Manage services
+      description: Services operations, like start/stop/restart a service/services, auto-start default services, and replan.
+      tags:
+        - services
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                action:
+                  type: string
+                  description: The action to perform. One of "start", "stop", "restart", "replan", "autostart".
```

Comment:
No need to repeat the possible values in the description -- let's just keep these in the enum.